src/argilla/server/contexts/accounts.py

```diff
@@ -141,6 +142,31 @@ async def create_user(db: "AsyncSession", user_create: UserCreate) -> User:
     return user
 
 
+async def create_user_with_random_password(
+    db,
+    username: str,
+    first_name: str,
+    workspaces: List[str] = None,
+    role: UserRole = UserRole.annotator,
+    password_length: int = 32,
+) -> User:
+    password = generate_random_password(password_length)
+
+    user_create = UserCreate(
+        first_name=first_name, username=username, role=role, password=password, workspaces=workspaces
+    )
+    return await create_user(db, user_create)
```

Comment:
If you think that `password_length` is not necessary:

```suggestion
async def create_user_with_random_password(
    db,
    username: str,
    first_name: str,
    workspaces: List[str] = None,
    role: UserRole = UserRole.annotator,
) -> User:    
    return await create_user(
        db,
        UserCreate(
            first_name=first_name,
            username=username,
            password=generate_random_password(),
            role=role,
            workspaces=workspaces,
        ),
    )
```